base/src/main/java/com/smartdevicelink/managers/screen/BaseTextAndGraphicManager.java

```diff
@@ -742,7 +742,7 @@ private boolean sdlArtworkNeedsUpload(SdlArtwork artwork){
 	}
 
 	private boolean shouldUpdatePrimaryImage() {
-		if (defaultMainWindowCapability == null || defaultMainWindowCapability.getImageTypeSupported() == null || defaultMainWindowCapability.getImageTypeSupported().size() > 0) {
+		if (defaultMainWindowCapability.getImageFields() == null || defaultMainWindowCapability.getImageTypeSupported() == null || defaultMainWindowCapability.getImageTypeSupported().size() > 0) {
```

Comment:
`defaultMainWindowCapability` may not be set and could be null if we get `onError` on line 200 or if we don't get a response at all. That applies to all other places where `defaultMainWindowCapability` is used.